pkg/provider/azure_vmss_cache.go

```diff
@@ -294,24 +313,36 @@ func (ss *ScaleSet) deleteCacheForNode(nodeName string) error {
 	return nil
 }
 
-func (ss *ScaleSet) newAvailabilitySetNodesCache() (*azcache.TimedCache, error) {
+func (ss *ScaleSet) newNonVmssUniformNodesCache() (*azcache.TimedCache, error) {
```

Comment:
what does this func name(`newNonVmssUniformNodesCache`) mean?

Reply:
it's vmss flex + avset, why vmss uniform is excluded? is `vmss uniform` logic more similar to vmss flex?

Reply:
newNonVmssUniformNodesCache contains vmss flex vm and avset vms.

The cache newNonVmssUniformNodesCache is used to replace the old availabilitySetNodesCache to distinguish the node type based on the nodeName.

One Key difference between vmssflex and vmss uniform is: in Flex, each VM is a top-level azure resource; while in Uniform, each VM is a sub-resource of VMSS. This means, the VM List API can only return VMSS Flex VMs, but it cannot return VMSS Uniform VMs. We utilize this feature to distinguish Flex VM and Uniform VM. The logic is:

- Use VM List API to retrieve all Visible VMs inside the resource group. The visible VMs include Flex VMs and AvSet VMs.
- for each returned VM, check whether it has a VMSS ID. If it has, this means it is a Flex VM. If not, this means it is an AvSet VM
- When checking a node type by NodeName, we first check if this node is inside the nonVmssUniformNodesCache. If not, it means it is a Uniform VM. If it is inside the nonVmssUniformNodesCache, then further check if the node has VMSS ID or not. If it has Vmss ID, it means it is FLex VM. If not, it is AvSet VM

Reply:
There indeed is a sperate cache for VMSS Flex nodes: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure_vmssflex.go#L53

The NonVmssUniformNodesCache inside azure_vmss.go is only used to distinguish Flex, Uniform and AvSet nodes. VM perperties are not retrieved from NonVmssUniformNodesCache